File: irrgarten/src/irrgarten/FuzzyPlayer.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;
import java.util.ArrayList;

/**
 * @author Juan Molina Molina
 * @author Paula Jiménez Ruiz
 */
class FuzzyPlayer extends Player{
        
    public FuzzyPlayer(Player other) {  
    
        super(other); 
    }
    
    @Override
    public Directions move(Directions directions, ArrayList<Directions> validMoves) {
        
        Dice dice=new Dice();
        Directions preferredDirection=super.move(directions, validMoves);
        return dice.nextStep(preferredDirection, validMoves, super.getIntelligence());
    }
    
    @Override
    public float attack() {
    
        return super.attack();
    }
    
    @Override
    protected float defensiveEnergy() {
    
        return super.defensiveEnergy();
    }
    
    @Override
    public String toString() {
    
        return "Fuzzy"+super.toString();
    }
}

File: ssh/FuzzyPlayer.rb
#encoding:utf-8

require_relative 'Player'

module Irrgarten

	class FuzzyPlayer < Player
	
		protected
			
			def defensive_energy
				return super
			end
			
		public
			
			def initialize (o)
				copy(o)
			end
			
			def move(directions, val_moves)
				pref_direction=super(directions,val_moves)
				return Dice::next_step(pref_direction,val_moves,get_intelligence)
			end
			
			def attack
				return super
			end
			
			def to_s
				return "Fuzzy" + super 
			end
			
	end
end
